pkg/controller/controller_create_test.go

```diff
@@ -0,0 +1,85 @@
+package controller
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/ablarry/golang-bootcamp-2020/pkg/model"
+	"github.com/ablarry/golang-bootcamp-2020/pkg/repo"
+	"github.com/ablarry/golang-bootcamp-2020/pkg/service"
+	"github.com/ablarry/golang-bootcamp-2020/pkg/utils/errors"
+	"github.com/gorilla/mux"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+// TestCreate test Create mocking service in controller
+func TestCreate(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	table := []struct {
+		name string
+		*http.Request
+		code        int
+		mockService func(c *gomock.Controller) *service.MockService
+		params      map[string]string
+		expected    *model.Pokemon
+		ok          bool
+	}{
+		{
+			"1._ OK: Create",
+			newRequest("POST", "/v1/pokemons", `{ "id": 1,"category": "Seed",  "name": "Bulbasaur", "type": "Grass", "wakness": "Fire" }`),
```

Comment:
For this and all other repetitions.

Multiple repetitions of the same value (in any kind of variables) is not encouraged when using Go.

Best practices suggest to move them out as constants and use them instead of an explicit value.

- Instead of `"POST"`, you could either declare it as a constant or use the constant in the `net/http` package for that.
- "v1/pokemons" to a constant
- The serialized JSON, instead of using as it is, you could have it stored in a file, and use a function to convert it to string as a variable; if you decide to add more stuff in the future and have a scenario with a really big JSON will cause troubles with line too long (some linters cry for that).